src/drivers/uploadthing.ts

```diff
@@ -0,0 +1,90 @@
+import { defineDriver } from "./utils";
+import { ofetch, $Fetch } from "ofetch";
+import { UTApi } from "uploadthing/server";
+
+export interface UploadThingOptions {
+  apiKey: string;
+}
+
+export default defineDriver<UploadThingOptions>((opts) => {
+  let client: UTApi;
+  let utFetch: $Fetch;
+
+  const getClient = () => {
+    return (client ??= new UTApi({
+      apiKey: opts.apiKey,
+      fetch: ofetch.native,
+    }));
+  };
+
+  const getUTFetch = () => {
+    return (utFetch ??= ofetch.create({
+      baseURL: "https://uploadthing.com/api",
+      headers: {
+        "x-uploadthing-api-key": opts.apiKey,
+      },
+    }));
+  };
+
+  return {
+    hasItem(key) {
+      // This is the best endpoint we got currently...
+      return getUTFetch()("/getFileUrl", {
+        body: { fileKeys: [key] },
+      }).then((res) => res.ok);
+    },
+    getItem(key) {
+      return ofetch(`https://utfs.io/f/${key}`);
+    },
+    getItemRaw(key) {
+      return ofetch
+        .native(`https://utfs.io/f/${key}`)
+        .then((res) => res.arrayBuffer());
+    },
+    getItems(items) {
+      return Promise.all(
+        items.map((item) =>
+          ofetch(`https://utfs.io/f/${item.key}`).then((res) => ({
+            key: item.key,
+            value: res,
+          }))
+        )
+      );
+    },
+    getKeys() {
+      return getClient()
+        .listFiles({})
```

Comment:
Just checking, if `listFiles` supports a base, `getKeys(base)` accepts it (and it can make usage faster specially when there are too many files and one needs to filter)

Reply:
uploadthing currently doesn't support directories, the only filter options this `listFiles` takes are pagination-related

Reply:
Oh okay i see. BTW unstorage doesn't too. We just assume `:` and `/` as fs-like seperators for prefixing. Would that be something you consider supporting at some point? (I imagine for quering large amount of files could come in use)

Reply:
Definitely something we would consider supporting